go/plugins/postgresql/docstore.go

```diff
@@ -24,6 +31,47 @@ type docStore struct {
 	embedderOptions any
 }
 
+
+// newDocStore instantiate a docStore
+func newDocStore(ctx context.Context, p *Postgres, cfg Config) (*docStore, error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	if !p.initted {
+		panic("postgres.Init not called")
+	}
+	if cfg.Name == "" {
+		return nil, fmt.Errorf("name is empty")
+	}
+	if cfg.Embedder == nil {
+		return nil, fmt.Errorf("embedder is required")
+	}
+
+	if cfg.SchemaName == "" {
+		cfg.SchemaName = defaultSchemaName
+	}
+
+	if cfg.TableName == "" {
+		cfg.TableName = defaultTable
+	}
+
+	return &docStore{
+		engine: p.engine,
+
+		tableName:             cfg.TableName,
+		schemaName:            cfg.SchemaName,
+		idColumn:              cfg.IDColumn,
+		metadataJSONColumn:    cfg.MetadataJsonColumn,
+		contentColumn:         cfg.ContentColumn,
+		embeddingColumn:       cfg.EmbeddingColumn,
+		metadataColumns:       cfg.MetadataColumns,
+		ignoreMetadataColumns: cfg.IgnoreMetadataColumns,
```

Comment:
See comments on retriever to remove `ignoreMetadataColumns: cfg.IgnoreMetadataColumns`